integration_test/runner/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+add_executable(integration_test.runner)
+emil_build_for(integration_test.runner HOST Windows)
+
+target_sources(integration_test.runner PRIVATE
+    FixtureEcho.cpp
+    FixtureEcho.hpp
+    Hooks.cpp
+    StepsGpio.cpp
+    StepsUart.cpp
+    Waiting.cpp
+    Waiting.hpp
+)
+
+target_link_libraries(integration_test.runner PRIVATE
+    args
+    cucumber-cpp-runner
+    integration_test.logic
+    hal.generic
+    gtest
+    gmock
+    services.network_instantiations
+)
+
+if (EMIL_BUILD_WIN)
+    target_link_libraries(integration_test.runner PRIVATE hal.windows)
+endif()
+
+if (EMIL_BUILD_UNIX OR EMIL_BUILD_DARWIN)
+    target_link_libraries(integration_test.runner PRIVATE hal.unix)
+endif()
```

Comment:
you can also depend on `hal.generic`. That dependency handles the `hal.windows` and `hal.unix` for you. It also comes with `UartGeneric`. Which abstracts away the OS specific part.